src/keys.rs

```diff
@@ -256,3 +256,63 @@ impl<C: CipherSuite> GroupVerifyingKey<C> {
         }
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::testing::Secp256k1Sha256;
+    use crate::utils::Scalar;
+    use crate::{FromBytes, ToBytes};
+    use ark_ff::UniformRand;
+    use rand::{rngs::OsRng, RngCore};
+
+    #[test]
+    fn test_serialization() {
+        let mut rng = OsRng;
+
+        for _ in 0..100 {
+            let skey = DiffieHellmanPrivateKey(Scalar::<Secp256k1Sha256>::rand(&mut rng));
+            let bytes = skey.to_bytes().unwrap();
+            assert!(DiffieHellmanPrivateKey::<Secp256k1Sha256>::from_bytes(&bytes).is_ok());
+            assert_eq!(
+                skey,
+                DiffieHellmanPrivateKey::<Secp256k1Sha256>::from_bytes(&bytes).unwrap()
+            );
+
+            let pkey = DiffieHellmanPublicKey::<Secp256k1Sha256>::new(
+                <Secp256k1Sha256 as CipherSuite>::G::generator().mul(skey.0),
+            );
+            let bytes = pkey.to_bytes().unwrap();
+            assert!(DiffieHellmanPublicKey::<Secp256k1Sha256>::from_bytes(&bytes).is_ok());
+            assert_eq!(
+                pkey,
+                DiffieHellmanPublicKey::<Secp256k1Sha256>::from_bytes(&bytes).unwrap()
+            );
+
+            let skey = IndividualSigningKey::<Secp256k1Sha256> {
+                index: rng.next_u32(),
```

Comment:
A random comment, unrelated to this PR: is it ok to have `index` be a plain `u32`, as opposed to `std::num::Wrapping` or `std::num::Saturating`? Could we avoid allowing users to set the index? If we can't, can we make them use a setter you think? Not sure what the risk would be but it seems bad to allow unexpected wrap here.

Reply:
The index of the signing key is equal to the index of the participant in the list.
That is, for a `t-out-of-n` scheme, the index will be `index <= n`.

I doubt that we need to have `n > 4B` and as such, we stuck with `u32` all around to save on some serialization size.

Also, I'm not sure why we'd get any saturating / wrapping operation here? We don't do any arithmetic operation on the index, it's just randomly generated here for the sake of testing serialization.

Reply:
We are not doing any arithmetic here (or anywhere else afaik?) but it's a public value (ok, `pub(crate)`) so a clumsy dev *could* misuse it.

But yeah, now that I re-read this I don't think this concern is valid.